core/domain/email_manager.py

```diff
@@ -163,6 +163,28 @@ def log_new_error(*args, **kwargs):
 
 
 def _require_sender_id_is_valid(intent, sender_id):
+    """Only allow emails as specified in the config.
```

Comment:
My first question on reading this was: what config?

Instead, I suggest something along these lines: "Ensure that the sender ID is valid, based on the email's intent." 

This allows the reader to connect what the function does to the args passed into it. The rest of the explanation is great.